src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -1601,6 +1480,15 @@ private void RecordVarRead(LocalSymbol local)
             locInfo.LocalDefs.Add(locDef);
         }
 
+        private bool EvalStackHasLocal(LocalSymbol local)
+        {
+            var top = _evalStack.Last();
```

Comment:
No, the need for this call only arises when current stack is deeper than at declaration of the local, so it cannot be empty.